### PR TITLE
feat: introduce API migration instructions

### PR DESCRIPTION
**Pull request by pleku**
- lists out incompatible/changed API per each legacy component class
- lists out unsupported API per each legaacy component class and instructions how to migrate

Closes vaadin/flow-legacy-components#72

### CODE REVIEW

**Comment by magi42** [comment]
I'll go through the changes in this PR as well as try to fix other issues in the file as well.

**Review by DiegoCardoso — CHANGES_REQUESTED** [review]
Spelling issues I found so far.

**Review by magi42 — COMMENTED** [review]
Did a round of language review and fixed a number of issues, but now noticed that there's still some TODO. Maybe do another round later or give to Doug.

File: articles/flow/upgrading/legacy-component-pack.asciidoc [hunk 1]
```diff
@@ -7,28 +7,29 @@ layout: page
...
+The API, behavior, and even the DOM-structure of the components is made to mimic the existing components as closely as possible.
+By utilizing the legacy components, you will have less code and logic to migrate and can get it ready to compile and run sooner.
 
-The first version of the pack will consist of the following components: `HorizontalLayout`, `VerticalLayot`, `Panel` and `Label`.
-Each component either doesn't have a corresponding component available with Flow, or the behavior and API has changes so significantly, that it would require significant effort to modify the code to migrate it to Flow.
-By using the legacy component pack, for the forementioned components, only the imports need to be changed to be able to use almost all of the same component features.
+The first version of the pack consists of the following components: `HorizontalLayout`, `VerticalLayot`, `Panel`, and `Label`.
+Each component either does ot have a corresponding component available with Flow, or the behavior and API has changes so significantly that it would require much effort to modify the code to migrate it to Flow.
```
Comment (rolfsmeds) [inline articles/flow/upgrading/legacy-component-pack.asciidoc:16]:
```suggestion
These components either lack counterparts in Flow entirely, or have significantly different APIs, requiring a lot of modifications when migrating.
```

**Review by rolfsmeds — APPROVED** [review]

**Comment by magi42** [comment]
As there are "TODO" sections, converted to draft to prevent accidental merging.

**Comment by pleku** [comment]
> As there are "TODO" sections, converted to draft to prevent accidental merging.

@magi42 please note that this is not targeting `latest` branch but `feat/legacy-component-pack` so the idea is to merge it there. We'll make a PR to `latest` once everything is in place.

**Comment by magi42** [comment]
@pleku Ah right, OK then.

**Pull request merged**